PopupKit/Sources/PopupKit/Implementation/Views/NotificationPresenterView.swift

```diff
@@ -13,95 +13,105 @@ struct NotificationPresenterView: View {
-        .offset(y: verticalDragTranslation < .zero ? verticalDragTranslation : .zero)
-        .offset(x: horizontalDragTranslation < .zero ? horizontalDragTranslation : .zero)
-        .gesture(dragGesture)
         .onTapGesture(perform: onTap)
+        .gesture(dragGesture)
         .cornerRadius(10)
         .padding(.horizontal, 15)
         .padding(.top, safeAreaInsets.top)
+        .offset(offset)
+        .animation(.interactiveSpring(), value: offset)
         .transition(.move(edge: dismissEdge))
     }
 }
-
+private extension NotificationPresenterView {
+    func processGeometry(_ geometry: GeometryProxy) -> some View {
+        return Color.init(uiColor: .adamant.swipeBlockColor)
+            .cornerRadius(10)
+    }
+    func onTap() {
+        model.tapHandler?.value()
+        dismissAction()
+        dismissEdge = .top
+    }
+}
 private extension NotificationPresenterView {
     var dragGesture: some Gesture {
         DragGesture()
-            .onChanged {
-                if dragDirection == nil {
-                    dragDirection = abs($0.translation.height) > abs($0.translation.width) ? .vertical : .horizontal
+            .onChanged { value in
+                if dragDirection == nil || (abs(value.translation.width) <= 5 && abs(value.translation.height) <= 5) {
+                    detectDragDirection(value: value)
+                }
+                if dragDirection == .vertical && isTextLimited {
+                    dynamicHeight = max(0, min(value.translation.height, 30))
                 }
-                switch dragDirection {
-                case .vertical:
-                    verticalDragTranslation = $0.translation.height
-                case .horizontal:
-                    horizontalDragTranslation = $0.translation.width
-                case .none:
-                    break
+                if dragDirection == .vertical, value.translation.height < 0 {
+                    offset = CGSize(width: 0, height: value.translation.height)
+                } else if dragDirection == .horizontal, value.translation.width < 0 {
+                    offset = CGSize(width: value.translation.width, height: 0)
                 }
             }
-            .onEnded {
-                if $0.velocity.height < -100 || -$0.translation.height > minTranslationYForDismiss {
-                        dismissEdge = .top
-                    Task { dismissAction() }
-                } else if $0.velocity.width < -100 || $0.translation.width > minTranslationXForDismiss {
-                        dismissEdge = .leading
-                    Task { dismissAction() }
-                } else if $0.velocity.height > -100 || -$0.translation.height < minTranslationYForDismiss {
-                    withAnimation {
-                        horizontalDragTranslation = .zero
+            .onEnded { value in
+                if dragDirection == .vertical {
+                    if value.translation.height > 25 {
```

Comment:
There are two if checks: `if value.translation.height > 25` and `if value.translation.height < -30`. Are you sure there should be different absolute values